posthog/async_migrations/migrations/0001_events_sample_by.py

```diff
@@ -0,0 +1,139 @@
+from constance import config
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.sql.events import EVENTS_TABLE
+from posthog.async_migrations.definition import AsyncMigrationDefinition, AsyncMigrationOperation
+from posthog.constants import AnalyticsDBMS
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+from posthog.version_requirement import ServiceVersionRequirement
+
+TEMPORARY_TABLE_NAME = "temp_events_0001_events_sample_by"
+
+
+class Migration(AsyncMigrationDefinition):
+
+    description = "Events table migration for compatible sample by column."
+
+    posthog_min_version = "1.30.0"
+    posthog_max_version = "1.31.0"
+
+    service_version_requirements = [
+        ServiceVersionRequirement(service="clickhouse", supported_version=">=21.6.0,<21.7.0"),
+    ]
+
+    operations = [
+        AsyncMigrationOperation(
+            database=AnalyticsDBMS.CLICKHOUSE,
+            sql=f"""
+            CREATE TABLE IF NOT EXISTS {TEMPORARY_TABLE_NAME} ON CLUSTER {CLICKHOUSE_CLUSTER} AS {EVENTS_TABLE}
+            ENGINE = ReplacingMergeTree(_timestamp)
+            PARTITION BY toYYYYMM(timestamp)
+            ORDER BY (team_id, toDate(timestamp), cityHash64(distinct_id), cityHash64(uuid))
```

Comment:
non-blocking and off topic

Why do we use `toDate(timestamp)` here? Does this mean that we'd need to use `toDate(timestamp)` in our queries to take advantage of data skipping? Apologies for my naivety.

Reply:
This is currently the case in `master`:

https://github.com/PostHog/posthog/blob/fc4699db8fa7f0de83917e32038726f6d7187525/ee/clickhouse/sql/events.py#L40

So while I began to type out my guesses as to the reasoning, I've now deleted them as it's best to ask elsewhere.